elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLColumnProjection.java

```diff
@@ -80,9 +82,35 @@ default boolean canNest(Queryable source, MetaDataStore metaDataStore) {
         boolean canNest = verifier.verify(sql);
         if (! canNest) {
             LOGGER.debug("Unable to nest {} because {}", this.getName(), verifier.getLastError());
+
+            return false;
         }
 
-        return canNest;
+        List<Reference> references = new ExpressionParser(metaDataStore).parse(source, this);
+
+        ColumnContext ctx = ColumnContext
+                .builder()
+                .metaDataStore(metaDataStore)
+                .column(this)
+                .queryable(source)
+                .tableArguments(source.getArguments())
+                .alias(source.getSource().getAlias())
+                .build();
+
+        //Search for any join expression that contains $$column.  If found, we cannot nest
+        //because rewriting the SQL in the outer expression will lose the context of the calling $$column.
+        JoinExpressionExtractor.JoinResolver columnResolver = (expression, context) -> expression;
+
+        return ! (references.stream().anyMatch(reference -> {
+            Set<String> joinExpressions = reference.accept(new JoinExpressionExtractor(ctx, columnResolver));
+
+            return joinExpressions.stream().anyMatch(expression -> {
+                if (expression.contains(COL_PREFIX)) {
```

Comment:
may be use `$$column.args.` instead of COL_PREFIX which is just `$$column` as we support $$column.expr also.